app/models/session/sms.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class Session::Sms < Session
+  SMS_CODE_MIN_LENGTH = 4
+  validates :sms_code, uniqueness: true, presence: true, length: { minimum: SMS_CODE_MIN_LENGTH }, if: -> { published? }
```

Comment:
We should have a unique index. We should unlock the code when the session is closed or archived. I believe sometimes researchers want to archive an old version and launch a new one without changing the code in the leaflets

Reply:
What to you think about conditional postgres index? It will ensure better protection with original idea of conditional uniqness.

Reply:
that's exactly what I'm thinking about 🚀